Test/Zip.Tests.ps1

```diff
@@ -0,0 +1,571 @@
+    Path:
+    - dir1\subdir\
+    Include:
+    - "*.txt"
+'@
+    ThenArchiveShouldInclude 'dir1\subdir\file.txt'
+}
+
+Describe 'Zip.when packaging a filtered directory' {
+    Init
+    GivenARepositoryWIthItems 'dir1\subdir\file.txt','dir1\one.ps1','dir1\dir2\file.txt'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    Path:
+    - dir1\subdir\
+    Include:
+    - "*.txt"
+    Exclude:
+    - dir2
+'@
+    ThenArchiveShouldInclude 'dir1\subdir\file.txt'
+    ThenArchiveShouldNotInclude 'dir1\one.ps1','dir1\dir2\file.txt'
+}
+
+Describe 'Zip.when packaging a directory with a space' {
+    Init
+    GivenARepositoryWIthItems 'dir 1\sub dir\file.txt'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    Path:
+    - dir 1\sub dir\
+    Include:
+    - "*.txt"
+'@
+    ThenArchiveShouldInclude 'dir 1\sub dir\file.txt'
+}
+
+Describe 'Zip.when packaging a directory with a space and trailing backslash' {
+    Init
+    GivenARepositoryWIthItems 'dir 1\sub dir\file.txt'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    Path:
+    - dir 1\sub dir\
+    Include:
+    - "*.txt"
+'@
+    ThenArchiveShouldInclude 'dir 1\sub dir\file.txt'
+}
+
+Describe ('Zip.when compression level is Optimal') {
+    Init
+    GivenARepositoryWithItems 'one.ps1'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    CompressionLevel: Optimal
+    Path:
+    - "*.ps1"
+'@
+    ThenArchiveShouldBeCompressed 'Zip.zip' -LessThanOrEqualTo 2800
+}
+
+Describe ('Zip.when compression level is Fastest') {
+    Init
+    GivenARepositoryWithItems 'one.ps1'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    CompressionLevel: Fastest
+    Path:
+    - "*.ps1"
+'@
+    ThenArchiveShouldBeCompressed 'Zip.zip' -GreaterThan 2800
+}
+
+Describe ('Zip.when compression level is NoCompression') {
+    Init
+    GivenARepositoryWithItems 'one.ps1'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    CompressionLevel: NoCompression
+    Path:
+    - "*.ps1"
+'@
+    ThenArchiveShouldBeCompressed 'Zip.zip' -GreaterThan (Get-Item -Path $PSCommandPath).Length
+}
+
+Describe 'Zip.when compression level is not included' {
+    Init
+    GivenARepositoryWIthItems 'one.ps1'
+    WhenPackaging @"
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    Path:
+    - "*.ps1"
+"@
+    ThenArchiveShouldBeCompressed 'Zip.zip' -LessThanOrEqualTo 2800
+}
+
+Describe 'Zip.when a bad compression level is included' {
+    Init
+    GivenARepositoryWIthItems 'one.ps1'
+    WhenPackaging -ErrorAction SilentlyContinue @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    CompressionLevel: this is no good
+    Path:
+    - "*.ps1"
+'@
+    ThenTaskFails 'is an invalid compression level'
+}
+
+Describe 'Zip.when package has empty directories' {
+    Init
+    GivenARepositoryWithItems 'root.ps1','dir1\one.ps1','dir1\emptyDir2\text.txt'
+    GivenARepositoryWithItems 'dir1\emptyDir1' -ItemType 'Directory'

```

Comment:
Test case is not accurate as described because the function `GivenARepositoryWithItems` never implements the `ItemType` parameter, resulting in `dir1\emptyDir1` being created as a File rather than a Directory.